### PR TITLE
WIP: Feature/implement colorconvert tests

### PR DESCRIPTION
**Pull request by m-herrera**

### CODE REVIEW

File: tests/check/gsttiovxcolorconvert.c [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+#include "gst-libs/gst/tiovx/gsttiovxcontext.h"
+#include "test_utils.h"
+
+
+static const int kImageWidth = 640;
+static const int kImageHeight = 480;
+static const vx_df_image kTIOVXImageFormat = VX_DF_IMAGE_RGB;
+static const char *kGstImageFormat = "RGB";
+
+static const int kMinBuffers = 1;
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:80]: min must be 2

File: tests/check/gsttiovxcolorconvert.c [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+static const int kMaxBuffers = 4;
+
+static const gchar * test_pipelines[] = {
+    "videotestsrc is-live=true num-buffers=5 ! video/x-raw,format=RGB,width=1280,height=720 ! tiovxcolorconvert ! video/x-raw,format=NV12,width=1280,height=720 ! fakesink async=false",
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:84]: No need fo such a big image. Please use a smaller image.

File: tests/check/gsttiovxcolorconvert.c [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+}
+
+GST_START_TEST (test_bypass_on_same_caps)
+{
+  GstHarness *h;
+  GstBuffer *in_buf;
+  GstBuffer *out_buf;
+  const gchar *caps =
+      "video/x-raw,format=RGB,width=640,height=480,framerate=30/1";
+  const gsize size = 640 * 480;
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:142]: This size is not okay. You need to multiply by 3, which is RGB's bytes per pixel factor.

File: tests/check/gsttiovxcolorconvert.c [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+  initialize_tiovx_buffer_pool (&pool);
+  gst_buffer_pool_acquire_buffer (pool, &in_buf, NULL);
+
+  /* Define caps */
+  gst_harness_set_src_caps_str (h, in_caps);
+  gst_harness_set_sink_caps_str (h, out_caps);
+
+  /* Push the buffer */
+  ret = gst_harness_push (h, in_buf);
+  gst_harness_pull(h);
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:199]: Please check for the pulled buffer as well. You might check the video meta to have the right format.

File: tests/check/gsttiovxcolorconvert.c [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+  fail_if (GST_FLOW_OK != ret);
+
+  gst_harness_teardown (h);
+
+  /* cleanup */
+  if (tiovx_context) {
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:206]: No need for this extra check. Free directly

File: tests/check/gsttiovxcolorconvert.c [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+  const gchar *in_caps =
+      "video/x-raw,format=RGB,width=640,height=480,framerate=30/1";
+  const gchar *out_caps =
+      "video/x-raw,format=NV12,width=640,height=480,framerate=30/1";
+
+  GstTIOVXContext *tiovx_context = gst_tiovx_context_new ();
+
+  h = gst_harness_new ("tiovxcolorconvert");
+
+  initialize_tiovx_buffer_pool (&pool);
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:190]: Free the pool at the end of the test.

File: tests/check/gsttiovxcolorconvert.c [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+  /* cleanup */
+  gst_buffer_unref (out_buf);
+  gst_harness_teardown (h);
+}
+GST_END_TEST;
+
+GST_START_TEST (test_state_change)
+{
+  test_states_change_success (test_pipelines[TEST_RGB_TO_NV12]);
```
Comment (michaelgruner) [inline tests/check/gsttiovxcolorconvert.c:170]: Did you forget to check in this?

**Review by michaelgruner — CHANGES_REQUESTED** [review]

File: tests/check/gsttiovxcolorconvert.c (hunk 7 above)
Comment (m-herrera) [reply tests/check/gsttiovxcolorconvert.c:170]: This is implemented in a test_utils file, it is added by the base branch for this PR so it is not shown

**Pull request closed without merging**